### PR TITLE
#161348807 Pagination support for articles

### PR DESCRIPTION
**Pull request by Sulaiman-Mozes**
#### What does this PR do?
  - It supports pagination for articles

#### Description of Task to be completed?
 - Ten articles by default per page on the list articles view
 - A link is provided to the next page
 - A link is provided to the previous page (if you have moved to the page)

#### How should this be manually tested?
On the following Endpoint below
 - GET /api/articles

If there are more than ten articles, the articles will be paginated and only ten to be displayed per page
 ````
{
  "article": {
    "count": 30,
    "next": "http://127.0.0.1:8000/api/articles?page=3",
    "previous": "http://127.0.0.1:8000/api/articles",
    "results": [
      {
        "id": 20,
        "slug": "string1541662928452356",
        "title": "string",
        "description": "string",
        "body": "string",
        "tag_list": [
          "string"
        ],
        "created_at": "2018-11-08T07:42:08.463039Z",
        "updated_at": "2018-11-08T07:42:08.463060Z",
        "favorited": false,
        "favorites_count": 0,
        "image_url": "http://www.vsh.bhsudj.urj",
        "audio_url": "http://wwww.buesj.buejn.buenwe",
        "author": {
          "id": 1,
          "username": "Sulaiman",
          "email": "sulaiman.kayizzi@andela.com"
        }
      },
      {
        "id": 19,
        "slug": "string1541662928125547",
        "title": "string",
        "description": "string",
        "body": "string",
        "tag_list": [
          "string"
        ],
        "created_at": "2018-11-08T07:42:08.145665Z",
        "updated_at": "2018-11-08T07:42:08.145690Z",
        "favorited": false,
        "favorites_count": 0,
        "image_url": "http://www.vsh.bhsudj.urj",
        "audio_url": "http://wwww.buesj.buejn.buenwe",
        "author": {
          "id": 1,
          "username": "Sulaiman",
          "email": "sulaiman.kayizzi@andela.com"
        }
      }, 
....
.....
````

#### What are the relevant pivotal tracker stories?
[#161348807](https://www.pivotaltracker.com/story/show/161348807)

### CODE REVIEW

File: authors/apps/articles/tests_articles/tests.py [hunk 1]
```diff
@@ -86,6 +86,15 @@ def test_get_all_articles(self):
         self.assertIn("How to train your Dragon", str(response.data))
         self.assertEqual(len(articles), 1)
 
+    def test_get_two_articles(self):
```
Comment (emugaya) [inline authors/apps/articles/tests_articles/tests.py:89]: Wrong test. Why do we get two articles?

File: authors/settings.py [hunk 2]
```diff
@@ -161,10 +161,13 @@
 AUTH_USER_MODEL = 'authentication.User'
 
 REST_FRAMEWORK = {
+    'DEFAULT_PAGINATION_CLASS':
```
Comment (emugaya) [inline authors/settings.py]: no need for `CLASS` in name

**Review by emugaya — CHANGES_REQUESTED** [review]
- How do we know the current page, total number of pages, previous e.t.c.
- Follow commit message best practices.
- Your PR description is no clear on what is returned.

File: authors/apps/articles/tests_articles/tests.py [hunk 3]
```diff
@@ -150,3 +159,15 @@ def test_delete_article(self):
                                       format='json')
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertIn("Article deleted sucessfully", str(response.data))
+
+    def test_invalid_page_number(self):
+        self.test_create_article()
+        articles_url = '/api/articles?page=3'
+        response = self.client.get(articles_url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
```
Comment (judeinno) [inline authors/apps/articles/tests_articles/tests.py:167]: I see the changes requested for in the call have been implemented

**Review by judeinno — APPROVED** [review]
Ready for merge

**Review by RachaelNantale — APPROVED** [review]
LGTM

File: authors/settings.py [hunk 4]
```diff
@@ -161,10 +161,13 @@
...
+        'rest_framework.pagination.PageNumberPagination',
+    'PAGE_SIZE': 10,
```
Comment (brucemakallan) [inline authors/settings.py:166]: I think this minimum value is fine. I suggest we leave it as it is.

**Review by brucemakallan — APPROVED** [review]

**Review by emugaya — APPROVED** [review]

**Pull request merged**